monitoring/resourse_lakehouse_monitor.go

```diff
@@ -0,0 +1,139 @@
+	OutputSchemaName         string                                   `json:"output_schema_name"`
+	Schedule                 *catalog.MonitorCronSchedule             `json:"schedule,omitempty"`
+	Snapshot                 Snapshot                                 `json:"snapshot,omitempty"`
+	SkipBuiltinDashboard     bool                                     `json:"skip_builtin_dashboard,omitempty"`
+	SlicingExprs             []string                                 `json:"slicing_exprs,omitempty"`
+	TimeSeries               *catalog.MonitorTimeSeriesProfileType    `json:"time_series,omitempty"`
+	WarehouseId              string                                   `json:"warehouse_id,omitempty"`
+}
+
+type MonitorInfo struct {
+	Monitor
+	DriftMetricsTableName   string                    `json:"drift_metrics_table_name,omitempty"`
+	ProfileMetricsTableName string                    `json:"profile_metrics_table_name,omitempty"`
+	Status                  catalog.MonitorInfoStatus `json:"status,omitempty"`
+	TableName               string                    `json:"table_name,omitempty"`
+}
+
+func ResourceLakehouseMonitor() common.Resource {
+	monitorSchema := common.StructToSchema(
+		MonitorInfo{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			common.MustSchemaPath(m, "output_schema_name").Required = true
+
+			common.MustSchemaPath(m, "inference_log", "granularities").Required = true
+			common.MustSchemaPath(m, "inference_log", "granularities").Optional = false
+
+			common.MustSchemaPath(m, "inference_log", "problem_type").Required = true
+			common.MustSchemaPath(m, "inference_log", "problem_type").Optional = false
+
+			common.MustSchemaPath(m, "inference_log", "timestamp_col").Required = true
+			common.MustSchemaPath(m, "inference_log", "timestamp_col").Optional = false
+
+			common.MustSchemaPath(m, "inference_log", "prediction_col").Default = "prediction"
+
+			common.MustSchemaPath(m, "time_series", "granularities").Required = true
+			common.MustSchemaPath(m, "time_series", "granularities").Optional = false
+
+			common.MustSchemaPath(m, "time_series", "timestamp_col").Required = true
+			common.MustSchemaPath(m, "time_series", "timestamp_col").Optional = false
+
+			common.MustSchemaPath(m, "drift_metrics_table_name").Computed = true
+			common.MustSchemaPath(m, "profile_metrics_table_name").Computed = true
+			common.MustSchemaPath(m, "status").Computed = true
+
+			return m
+		},
+	)
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			tmpSchema := removeSnapshotField(monitorSchema)
+
+			var create catalog.CreateMonitor
+			common.DataToStructPointer(d, tmpSchema, &create)
+			if _, ok := d.GetOk("snapshot"); ok {
+				create.Snapshot = struct{}{}
+			}
+			create.FullName = d.Get("table_name").(string)
```

Comment:
Why do we need this? `DataToStructPointer` deserialize things - you already have `FullName` annotated as `table_name`...

Reply:
In the `catalog.CreateMonitor` struct we define the `FullName` with the json path `json:"-" url:"-"`. Therefore we have to set it manually as there is no json path that the schema can set it to.